README.md

```diff
@@ -1,83 +1,124 @@
+*   [Create a team](https://support.elastic.io/support/solutions/articles/14000048250-how-to-create-and-manage-a-team) to work on your new component. This is not required but will be automatically created using random naming by our system so we suggest you name your team accordingly.
+*   [Create a repository](https://support.elastic.io/support/solutions/articles/14000038797-create-and-manage-your-repositories) inside the team to deploy your new component code.
 
-## Examples of Node.js components
+### Examples of Node.js components
 
-We have numerous components build on Node.js apart from above mentioned [hello-world-nodejs](https://github.com/elasticio/hello-world-nodejs). Here are some  other community component examples:
+Here is a list of components build on Node.js:
 
-* [code-component](https://github.com/elasticio/code-component) to run pieces of synchronous JavaScript inside your integration flow,
-* [webhook-component](https://github.com/elasticio/webhook-component) to send and receive WebHooks on elastic.io platform,
-* [csv-component](https://github.com/elasticio/csv-component) to work with CSV files in your integration flow,
-* [sugarcrm-component](https://github.com/elasticio/sugarcrm-component) to use Sugar CRM in your integrations
-* and many more ...
+*   [petstore-component-nodejs](https://github.com/elasticio/petstore-component-nodejs) to build your first component
+*   [code-component](https://github.com/elasticio/code-component) to run pieces of synchronous JavaScript inside your integration flow,
+*   [webhook-component](https://github.com/elasticio/webhook-component) to send and receive WebHooks on elastic.io platform,
+*   [csv-component](https://github.com/elasticio/csv-component) to work with CSV files in your integration flow,
+*   [sugarcrm-component](https://github.com/elasticio/sugarcrm-component) to use Sugar CRM in your integrations
 
 [travis-image]: https://travis-ci.org/elasticio/sailor-nodejs.svg?branch=master
 [travis-url]: https://travis-ci.org/elasticio/sailor-nodejs
 [daviddm-image]: https://david-dm.org/elasticio/sailor-nodejs.svg?theme=shields.io
 [daviddm-url]: https://david-dm.org/elasticio/sailor-nodejs
 
+
+
 ## Sailor hooks
 
+### Init hook
+
+```javascript
+/**
+* cfg - config object with credentials and other stuff like datasamples
+*/
+exports.init = function(cfg) {
+    //do stuff
+    return Promise.resolve();
+}
+```
+
 ### Startup hook
- - Only on the first trigger
- - Called without ``this``
- - May return promise
- - May return value
- - May throw - not recommended
- - May return a promise that will fail
 
- - TBD - Startup logs, where to find it?
- - TBD - Separate them under different tab in UI
- - TBD - Where to see restart errors?
+```javascript
+/**
+* cfg - config object with credentials and other stuff like datasamples
+*/
+exports.startup = function(cfg) {
+    //do stuff
+    const data = messages.newMessageWithBody({
```

Comment:
Pls remove `messages.newMessageWithBody`,
just return an object with data in the promise.